src/util/system.cpp

```diff
@@ -6,7 +6,16 @@
 #include <util/system.h>
 
 #ifdef ENABLE_EXTERNAL_SIGNER
+#if defined(__GNUC__) && !defined(__clang__)
+// Boost 1.78 requires the following workaround.
+// See: https://github.com/boostorg/process/issues/235
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnarrowing"
+#endif
```

Comment:
It does not work on macOS.

Reply:
Also the similar changes should be applied to `src/test/system_tests.cpp`.

Reply:
According to [docs](https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-via-pragmas) clang processes `pragma GCC`, so just dropped `&& !defined(__clang__)`. Also added in the test file. I wonder if we should add a header file to avoid repeating this.